### ISSUE TITLE
va-modal: add aria-labelledby prop

### ISSUE THREAD

**Issue by powellkerry** [department-of-veterans-affairs/vets-design-system-documentation#2189]
## Description
va-modal needs to accept an `aria-labelledby` property so that teams can choose a heading/element that labels the modal. 

## Details
Currently the `aria-label` is set when the modal-title is defined, however there is an option to have no title in the modal. For this case there needs to be a way to set the `aria-labelledby` attribute so that screen readers can read the context of the modal.

See the originating comment here: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2087#issuecomment-1747393639

## Tasks
- [ ] Add new `aria-labelledby` property
- [ ] Ensure that logic is updated so that if both `modal-title` and `aria-labelledby` are defined that screen readers don't read both/repeat

## Acceptance Criteria
- [ ]

**Comment by laflannery** [comment]
@powellkerry I just want to clarify that the prop mentioned above has a typo - this should be `aria-labelledby` with 2 L's
